Hashchange: Change hash of #all-messages to #feed.

nimishmedatwal opened the pull request:
This pr makes hashchange of #all-messages to #feed, also redirecting users from `#all-messages` to `#feed` along with changing hash of help docs from `/all-messages` to `/combined-feed` along with it's references.
[CZO Discussion](https://chat.zulip.org/#narrow/stream/101-design/topic/All.20messages.20view.20label.20and.20icon/near/1668230)
<!-- Describe your pull request here.-->

Fixes: #27802.

<!-- If the PR makes UI changes, always include one or more still screenshots to demonstrate your changes. If it seems helpful, add a screen capture of the new functionality as well.

Tooling tips: https://zulip.readthedocs.io/en/latest/tutorials/screenshot-and-gif-software.html
-->

**Screenshots and screen captures:**

https://github.com/zulip/zulip/assets/91622060/17683840-e004-4f63-b386-39cfe75490d9

![image](https://github.com/zulip/zulip/assets/91622060/c0528b47-e4e1-4c92-b8b0-a10b28f69ed7)
![image](https://github.com/zulip/zulip/assets/91622060/53803808-72f6-4eaf-abc3-289267d05590)


<details>
<summary>Self-review checklist</summary>

<!-- Prior to submitting a PR, follow our step-by-step guide to review your own code:
https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code -->

<!-- Once you create the PR, check off all the steps below that you have completed.
If any of these steps are not relevant or you have not completed, leave them unchecked.-->

- [x] [Self-reviewed](https://zulip.readthedocs.io/en/latest/contributing/code-reviewing.html#how-to-review-code) the changes for clarity and maintainability
      (variable names, code reuse, readability, etc.).

Communicate decisions, questions, and potential concerns.

- [x] Explains differences from previous plans (e.g., issue description).
- [x] Highlights technical choices and bugs encountered.
- [x] Calls out remaining decisions and concerns.
- [x] Automated tests verify logic where appropriate.

Individual commits are ready for review (see [commit discipline](https://zulip.readthedocs.io/en/latest/contributing/commit-discipline.html)).

- [x] Each commit is a coherent idea.
- [x] Commit message(s) explain reasoning and motivation for changes.

Completed manual review and testing of the following:

- [x] Visual appearance of the changes.
- [x] Responsiveness and internationalization.
- [x] Strings and tooltips.
- [x] End-to-end functionality of buttons, interactions and flows.
- [x] Corner cases, error conditions, and easily imagined bugs.
</details>

nimishmedatwal commented:
@zulipbot add "area: left sidebar (ui)", "area: documentation (user)"

alya commented:
@laurynmm Are you up for reviewing this one? I haven't tested.

laurynmm commented:
@nimishmedatwal - Thanks for working on this renaming!

I think it would be good to split these changes into two commits. One for the URL update in the web-app and one for the help center changes.

For the help center changes, you're going to need to add a URL redirect for the renaming of the "All messages" article markdown file. See `zerver/lib/url_redirects.py` and some of the pull requests that have updated that for help center changes.

For now, I'm removing the "maintainer review" label. Let me know if you have any questions and ping me again for a review when you've updated for those changes.

alya commented:
@nimishmedatwal You can use Zulipbot to re-add the maintainer review label once this PR passes tests.

nimishmedatwal commented:
@zulipbot add "maintainer review"

laurynmm reviewed (commented):
@nimishmedatwal - Thanks for separating these in to two commits. It was much easier to review that way.

I had a few comments on the URL hash commit. Let me know if you have any questions about those. You should also check the server code to see if there are any remaining references to the `#all_messages` hash.

The help center updates look good. I was thinking we should also probably rename the `help/include/go-to-all-messages.md` in this commit.

One other thing I noticed is that we've already updated the instruction for the mobile app to use Combined feed, but we haven't yet updated that in the app. It still says All messages. @alya - What are your thoughts on that change? Is there a mobile issue for updating this string?

nimishmedatwal commented:
@laurynmm Thanks for the thorough review!
I've corrected all the issues you pointed out, also should I change `user_settings.web_home_view` in server? For now I've changed it back to `all_messages`. 
Also I am not sure whether to change this instance of `all_messages` in `help_relative_link.py`
```python
message_info = {
    "drafts": ["Drafts", "/#drafts", draft_instructions],
    "scheduled": ["Scheduled messages", "/#scheduled", scheduled_instructions],
    "recent": ["Recent conversations", "/#recent", recent_instructions],
    "all": ["Combined feed", "/#all_messages", all_instructions],
    "starred": ["Starred messages", "/#narrow/is/starred", starred_instructions],
    "direct": ["Direct message feed", "/#narrow/is/dm", direct_instructions],
    "inbox": ["Inbox", "/#inbox", inbox_instructions],
}
```

timabbott commented on web/src/hash_parser.ts:
```diff
@@ -119,7 +119,7 @@ export function is_spectator_compatible(hash: string): boolean {
         "message-formatting",
         "search-operators",
-        "all_messages",
+        "feed",
```
@nimishmedatwal please use the `TODO/compatibility` syntax and don't skip words; what you've written isn't proper English.

timabbott commented on web/src/hashchange.js:
```diff
@@ -199,6 +199,15 @@ function do_hashchange_normal(from_reload) {
         case "#all_messages":
+            // "#all_messages" was deprecated and replaced by
+            // "#feed" in 2024 so if a user goes to "#all_messages"
+            // (through saved boookmark or any such scenario) they'll
```
`boookmark` is not spelled correctly. I highly recommend using a tool like `codespell` to spell check your code, but also taking more time to be careful in writing -- it's one of the most important parts of a project like this.

timabbott commented on web/src/hotkey.js:
```diff
@@ -172,7 +172,7 @@ const keypress_mappings = {
     85: {name: "mark_unread", message_view_only: true}, // 'U'
     86: {name: "view_selected_stream", message_view_only: false}, // 'V'
-    97: {name: "all_messages", message_view_only: true}, // 'a'
+    97: {name: "feed", message_view_only: true}, // 'a'
```
`combined_feed` would be a much better name for this use case.

timabbott commented on help/marking-messages-as-read.md:
```diff
@@ -22,8 +22,8 @@ web/desktop app.
-  to preview new messages in a stream, or skim [All
-  messages](/help/all-messages), and later read each topic in detail.
+  to preview new messages in a stream, or skim [Combined
+  feed](/help/combined-feed), and later read each topic in detail.
```
I think "skim your [combined feed]" would be a better way to write this sentence given the new language -- the "your" feels more natural here than with the previous name.

timabbott commented:
> I've corrected all the issues you pointed out, also should I change `user_settings.web_home_view` in server?

Yes, but please do that in its own commit, and include a migration to update the value for all existing users and realmuserdefaults. That can possibly be done in a follow-up PR.

timabbott commented:
For that `help_relative_link` detail, that whole section seems like dead code; @drrosa may remember the context (0415e66969840aa450e9c06407921547bd4c3b93 removed the only use of it).
```
$ git grep 'relative\|message\|' help
help/direct-messages.md:{relative|message|direct}
```

timabbott commented:
I went ahead and did the finishing work on this to make sure I was happy with the strings in the comments; with those changes, I think this is good to merge, thanks @nimishmedatwal and @laurynmm! @nimishmedatwal please take more time in your PRs to be careful -- read the surrounding code, run `git log` on the files you're changing, and `git log -S` on identifiers you don't understand -- and double-check your English writing -- it can really help reduce errors. I also ended up rewriting some of your commit messages. 

For follow-ups, we still need an API change and database migration to close this out -- check the history for the "recent topics" rename to see how to do that. And then we also will want to rename variables -- e.g., `git grep all_messages_view` finds a function to rename.

nimishmedatwal commented:
@timabbott sure! I will keep that in mind in future.

laurynmm commented:
> For follow-ups, we still need an API change and database migration to close this out -- check the history for the "recent topics" rename to see how to do that.

Just a note here that we haven't yet done the API and database migration change for the "recent topics" rename yet, see issue #25781.
